sdk/core/Azure.Core/tests/TestFramework/RecordSession.cs

```diff
@@ -93,7 +93,11 @@ public bool IsEquivalent(RecordSession session, RecordMatcher matcher)
                 return false;
             }
 
-            return session.Variables.SequenceEqual(Variables) &&
+            // The DateTimeOffsetNow variable is updated any time it's used so
+            // we only care that both sessions use it or both sessions don't.
+            var now = TestRecording.DateTimeOffsetNowVariableKey;
```

Comment:
Why not do the same thing ID reuse does where we get the value from the previous recording if available? Would avoid having this check.

Reply:
The service won't let us use DateTime values in the past so it always needs to be updated when we re-record.